src/utils/SortingNetworkGenerator.java

```diff
@@ -1,129 +1,266 @@
+
+        if (small) {
+            out.write("<svg width='" + w + "' height='" + h + "' xmlns='http://www.w3.org/2000/svg'>");
+            out.write(writer.getValue());
+        }
+        out.write("</svg>");
+        if (monitor != null) monitor.close();
+        return false;
+    }
+
+    public static boolean encodeNetwork(Comparator[] comparators, File file) {
+        try (PrintWriter out = new PrintWriter(
+                new BufferedWriter(
+                    new OutputStreamWriter(
+                        new FileOutputStream(file), StandardCharsets.UTF_8
+                    ), OUT_BUFFER_SIZE),
+                false)
+            ) {
+            boolean cancelled = encodeNetwork0(comparators, out);
+            if (cancelled) {
+                JOptionPane.showMessageDialog(null, "Sorting network visualization cancelled",
+                    "Sorting Network Visualizer", JOptionPane.INFORMATION_MESSAGE);
                 return false;
             }
-        }
-        catch (IOException e) {
-            JErrorPane.invokeErrorMessage(e, "Sorting Network Visualizer");
-            return false;
-        }
-        catch (Exception e) {
+        } catch (Exception e) {
             JErrorPane.invokeErrorMessage(e, "Sorting Network Visualizer");
             return false;
         }
         return true;
     }
 
-    public static String encodeNetworkAndDisplay(String name, Integer[] indices, int arrayLength) {
-        String path = "network_" + name + "_" + arrayLength + ".svg";
-        int[] indicesInt = new int[indices.length];
-        for (int i = 0; i < indices.length; i++) {
-            indicesInt[i] = indices[i];
+    public static String encodeNetworkAndDisplay(String name, ArrayList<Integer> indices, int arrayLength) {
+        Comparator[] comparators;
+        try {
+            comparators = new Comparator[indices.size() / 2];
+            for (int i = 0, j = 1; i < comparators.length; i++, j += 2) {
+                comparators[i] = new Comparator(indices.get(j - 1), indices.get(j));
+            }
+        } catch (OutOfMemoryError e) {
+            JErrorPane.invokeErrorMessage(e, "Sorting Network Visualizer");
+            return null;
         }
-        if (!encodeNetwork(indicesInt, path)) {
+        System.out.println("Length: " + arrayLength + "\tComparators: " + comparators.length / 2);
+        indices.clear();
+        indices.trimToSize();
+        String path = "network_" + name + "_" + arrayLength + ".svg";
+        File file = new File(path);
+        try {
+            if (!encodeNetwork(comparators, file)) {
+                return null;
+            }
+        } catch (OutOfMemoryError e) {
+            JErrorPane.invokeCustomErrorMessage(
+                "ArrayV ran out of memory trying to visualize this sorting network.\n" +
+                "Either run ArrayV with more memory (or a smaller maximum length) or contemplate your life choices."
```

Comment:
I feel like programmer humor is a great thing, and almost never out of place.

Reply:
I mean that I don't know of any other instance of such humor elsewhere in the program (of course it's very likely there's a line I haven't seen)

I guess I should clarify that it takes a pretty "dark" turn fast

Reply:
I would disagree that this is a "dark" turn. I mean, this is someone running something on a ludicrous number of elements. Not to mention that this line almost never runs unless the user has less than 8GB of RAM (which is actually likely because the JVM only allocates 4GB by default on Windows).

Reply:
still feel it feels kinda awkward, but I guess if you feel it's fine it's fine